### ISSUE TITLE
CIAM authorities cause a null ref when using confidential client scenarios (daemon)

### ISSUE THREAD

**Issue by trwalke** [AzureAD/microsoft-identity-web#2732]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

2.17.3

### Web app

Sign-in users

### Web API

Protected web APIs (validating tokens)

### Token cache serialization

In-memory caches

### Description

For confidential client scenarios in IdWeb, when using the instance and tenant id options (`MicrosoftIdentityOptions`) to configure a CIAM authority, the flow works fine, However, when using the "Authority" option with a CIAM tenant, it fails. This failure occurs with  https://DomainName.ciamlogin.com and https://DomainName.ciamlogin.com/tenant
Reproduced using the `ChallengeUser_MicrosoftIdFlow_LocalApp_ValidEmailPassword` test

### Reproduction steps

Reproduced using the `ChallengeUser_MicrosoftIdFlow_LocalApp_ValidEmailPassword` test.
1. Update client id to use CIAM app
2. Remove instance and tenant id options 
3. Add CIAM authority

### Relevant code snippets

```csharp
TokenAcquirerFactory tokenAcquirerFactory = TokenAcquirerFactory.GetDefaultInstance();
            IServiceCollection services = tokenAcquirerFactory.Services;

            services.Configure<MicrosoftIdentityApplicationOptions>(s_optionName, option =>
            {
                option.Authority = "https://{CiamDomain}.ciamlogin.com/";
                option.ClientId = "Ciam app id";
                option.ClientCredentials = s_clientCredentials;
            });

            await CreateGraphClientAndAssert(tokenAcquirerFactory, services);
```

### Expected behavior

Test should pass without error. 
(Deamon app calls graph)

**Comment by bgavrilMS** [comment]
Thanks @trwalke , given that CIAM non-CUD works, I'm marking this as a P3. 

For Id.Web to support other OIDC authorities, this needs to be fixed.

**Comment by trwalke** [comment]
> Thanks @trwalke , given that CIAM non-CUD works, I'm marking this as a P3.
> 
> For Id.Web to support other OIDC authorities, this needs to be fixed.

This is not the case, the issue above is using the non-CUD domain and it is failing, it also fails with the CUD domain

It appeared that way when testing from the sample because of the api the sample used, but testing the ID.Web api reveals that both forms are failing
![image](https://github.com/AzureAD/microsoft-identity-web/assets/30090357/658551ef-2a9e-43a7-ad1a-e887d1e1bfc3)

**Comment by bgavrilMS** [comment]
@trwalke - the current CIAM daemon sample advises to use the Authority as

![image](https://github.com/AzureAD/microsoft-identity-web/assets/12273384/e4deb0bf-4ac7-4914-95f3-d05dc06d2d0c)


https://github.com/Azure-Samples/ms-identity-ciam-dotnet-tutorial/blob/main/2-Authorization/3-call-own-api-dotnet-core-daemon/ToDoListClient/appsettings.json

I believe this works (and there are some tests to cover it in Id.Web)?

**Comment by trwalke** [comment]
> @trwalke - the current CIAM daemon sample advises to use the Authority as
> 
> ![image](https://private-user-images.githubusercontent.com/12273384/318513871-e4deb0bf-4ac7-4914-95f3-d05dc06d2d0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5OTIxMjQsIm5iZiI6MTcxMTk5MTgyNCwicGF0aCI6Ii8xMjI3MzM4NC8zMTg1MTM4NzEtZTRkZWIwYmYtNGFjNy00OTE0LTk1ZjMtZDA1ZGMwNmQyZDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDE3MTcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljNWI1NWVlYWMyNzdkNDgyZTg5OTIzNmU4OWYzMTU5ODE2NmMwNzBkODU2ODA3ZmUwNmVhM2M5ZDRmYzZmNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wziIKbjtQvL3DEkDiQoxMgDCeTQ4mGAo2WZHfpf_woU)
> 
> https://github.com/Azure-Samples/ms-identity-ciam-dotnet-tutorial/blob/main/2-Authorization/3-call-own-api-dotnet-core-daemon/ToDoListClient/appsettings.json
> 
> I believe this works (and there are some tests to cover it in Id.Web)?



> @trwalke - the current CIAM daemon sample advises to use the Authority as
> 
> ![image](https://private-user-images.githubusercontent.com/12273384/318513871-e4deb0bf-4ac7-4914-95f3-d05dc06d2d0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5OTIxMjQsIm5iZiI6MTcxMTk5MTgyNCwicGF0aCI6Ii8xMjI3MzM4NC8zMTg1MTM4NzEtZTRkZWIwYmYtNGFjNy00OTE0LTk1ZjMtZDA1ZGMwNmQyZDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDE3MTcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljNWI1NWVlYWMyNzdkNDgyZTg5OTIzNmU4OWYzMTU5ODE2NmMwNzBkODU2ODA3ZmUwNmVhM2M5ZDRmYzZmNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wziIKbjtQvL3DEkDiQoxMgDCeTQ4mGAo2WZHfpf_woU)
> 
> https://github.com/Azure-Samples/ms-identity-ciam-dotnet-tutorial/blob/main/2-Authorization/3-call-own-api-dotnet-core-daemon/ToDoListClient/appsettings.json
> 
> I believe this works (and there are some tests to cover it in Id.Web)?

Yes, it worked when I ran it in the sample but it does not appear to be working with the test. The test is using the options property directly instead of using the appsettings.json so maybe this makes a difference? not sure. 

Also, I dont see any other CIAM tests that test the higher level `CreateGraphClientAndAssert()` api except for mine. The other CIAM tests target very specific helper methods.

**Comment by bgavrilMS** [comment]
P2 since this is a null ref.

**Comment by jennyf19** [comment]
@bgavrilMS and @trwalke was this fixed?

**Comment by bgavrilMS** [comment]
Not a common scenario, won't fix.
